File: python/demos/demo_elastic_bands.py
#!/usr/bin/env python3
"""Elastic Bands demo."""

from __future__ import annotations

from demo_common import run_local

# Imported from the submodule directly, mirroring demo_dwa.py's precedent (the
# package re-export lives in navigation/local_planning/band/__init__.py).
from navigation.local_planning.band.elastic_bands import ElasticBandsPlanner

if __name__ == "__main__":
    run_local("elastic_bands", ElasticBandsPlanner)

File: cpp/demos/demo_elastic_bands.cpp
#include "demo_common.hpp"
#include "navigation/local_planning/band/elastic_bands.hpp"

int main(int argc, char** argv) {
  try {
    return demo::run_local(argc, argv, "elastic_bands", [](const navigation::core::ParamSet& p) {
      return navigation::local_planning::ElasticBandsPlanner(p);
    });
  } catch (const std::exception& e) {
    std::cerr << "error: " << e.what() << "\n";
    return 1;
  }
}
